Update sync design docs to include the selective sync (#1526)
Automattic/studio: docs/design-docs/sync.md

- `wp-config.php` file
- `meta.json` file

> Any folder is optional and will be ignored if it does not exist, allowing partial syncs. The Jetpack backup is not a destructive process, meaning that the existing files will not be deleted.

Reviewer: Maybe we could consider slight adjustment here:

```suggestion
Folders in the archive are optional - if any are missing, they will simply be ignored, allowing for partial syncs. The Jetpack backup process is non-destructive: existing files will not be deleted.
```

Reviewer 2: I am personally against using the em dash in docs as it presents a break in the thought process. We would like to have our documentation clear and concise.

Reviewer 3: > Any folder is optional and will be ignored if it does not exist, allowing partial syncs. 

This bit was not clear to me but I am unsure of what wording would make it better. I think what Ivan has is better but it is not 100% clear to me either.

Author: Changed here. I used a regular dot `.` instead of the dash.
https://github.com/Automattic/studio/pull/1526/commits/1a8c054a0d7c6df6f6788b8ad79988fd9204c9e2

### Pull

When the user clicks **Pull**, Studio opens a sync dialog that allows them to select exactly which data to synchronize from the remote site to their local environment.

> By default, all files and the database are pre-selected. However, users can customize this operation by opening a dropdown to select specific files and folders to pull. Then the file tree will be displayed with wp-content as the root folder. This enables selective syncing of specific site groups.

Reviewer: ```suggestion
By default, all files and the database are pre-selected. Users can customize the sync by opening a dropdown and selecting specific files and folders to pull. The file tree is then displayed with wp-content as the root, enabling selective syncing of specific parts of the site.
```

Reviewer 3: > Users can customize the sync by opening a dropdown and selecting specific files and folders to pull.

I would use maybe `sync options` instead of just `the sync` to make the language more clear.

Author: Updated! https://github.com/Automattic/studio/pull/1526/commits/f660c0d45137a48b8c7f970a6f04ba75c1a02a44
Sync options sounds better. I left the article `the` in following commits.

The available options are:

> - **wp-content**: Select this to perform a complete pull of all site content. This operation will overwrite the local folders with the contents from the backup, including the themes, plugins, and uploads folders. Other existing files and folders will be ignored and not overwritten.

Reviewer 3: ```suggestion
- **wp-content**: Select this to pull all of the content from your site. This operation will overwrite the local folders with the contents from the backup, including the themes, plugins, and uploads folders. Other existing files and folders will be ignored and not overwritten.
```

Author: Done https://github.com/Automattic/studio/pull/1526/commits/4f42e4594558f505f8e816f765ed87c33bf63164

  - **plugins**: Pull all plugins from the remote site. This replaces any local plugins folder with the new ones found in the backup.

>   - **themes**: Pull all themes from the remote site. Like plugins, this replaces any existing local themes folder with the new ones found in the backup.

Reviewer 3: ```suggestion
  - **themes**: Pull all themes from the remote site. This replaces any existing local themes folder with the new ones found in the backup.
```

I would skip the comparison word here just to avoid confusing the users and make it straight to the point

Author: Done https://github.com/Automattic/studio/pull/1526/commits/38a43d25de332a7b158b9a20e1c974cb45a12583

  - **uploads**: Pull the media library (wp-content/uploads) from the remote site. This option will overwrite the local site’s uploads directory with the files from the backup.
  - **Other files and directories**: Pull all other files and folders inside wp-content that are not covered by the options above (for example, mu-plugins, fonts, etc.). This ensures custom or less-common directories can also be selectively synchronized.
- **database**: Pull the database from the remote site. This will overwrite the local database with the one from the backup. The database will be updated by running the WP-CLI sqlite import command.

> When the user clicks "Pull," Studio sends a request to the WPcom API to create a Jetpack Backup and generate a download link. We poll the API until the process is complete, then download the backup file to a temporary folder. The backup file will contain only the selected options, then it will be extracted, and the requested changes will be applied locally. If a full sync is selected, the local site will be completely replaced, and the database will be updated. If only certain parts are selected, only those will be updated.

Reviewer: ```suggestion
When the user clicks **Pull**, Studio sends a request to the WPcom API to create a Jetpack Backup and generate a download link. We then poll the API until the backup is ready, download the file to a temporary folder, and extract its contents.

The backup file includes only the options selected by the user. Once extracted, the corresponding changes are applied locally. If a full sync is selected, the local site is completely replaced, including the database. If only specific parts are selected, only those are updated.
```

Reviewer 4: Let's remove the 'We' - it's not us doing this, it's Studio.

Author: Thanks for suggestions and marking the word in bold. I updated it here: https://github.com/Automattic/studio/pull/1526/commits/319a885ab2683060aa79b39852047b2a4a6d611c
I also replaced We with Studio 👍

### Push

When the user clicks **Push**, Studio opens a sync dialog allowing them to choose exactly which data to synchronize from their local site to the remote environment.

> By default, all files and the database are pre-selected. Users can customize what gets pushed by displaying a file tree with wp-content as the root folder, where they can select individual plugins, themes, uploads subfolders for a more granular sync.

Reviewer: Maybe we could use similar paragraph as for the pull, just replacing the word `pull` with `push`:

```suggestion
By default, all files and the database are pre-selected. Users can customize the sync by opening a dropdown and selecting specific files and folders to push. The file tree is then displayed with wp-content as the root, enabling selective syncing of specific parts of the site.
```

Reviewer 3: >  Users can customize the sync by opening a dropdown and selecting specific files and folders to push. 

Again, I would use `sync options` here.

Author: Updated. https://github.com/Automattic/studio/pull/1526/commits/c7b4a755d455a1b2604c169bb52f2e1cd810034f
Here is where I introduced the article *the sync options*

See the Pull section for more details on the available options.

> When the user initiates a push, Studio creates a backup archive containing only the files and database tables the user has selected. Studio then uploads this backup to the WPcom API, which restores the selected components on the remote site. The process is non-destructive, meaning that the existing files not included in the backup will not be deleted.

Reviewer: ```suggestion
When the user initiates the push, Studio creates a backup archive containing only the files and database tables the user has selected. Studio then uploads this backup to the WPcom API, which restores the selected components on the remote site. The process is non-destructive: existing files and data not included in the backup are left unchanged.
```

Author: Done! https://github.com/Automattic/studio/pull/1526/commits/be0904dcd882138ebdc6487600184b5732478d8b

> An email notification will be sent from the backend after the Push has finished, informing the user that the push is complete and which options were pushed.

Reviewer 3: ```suggestion
An email notification will be sent from the backend after the Push has finished, informing the user that the push is complete and which options were synchronized.
```

Author: I updated and rephrased the entire sentence to use active voice.

https://github.com/Automattic/studio/pull/1526/commits/8e010c229e5761fc900be375fca6d214f336a311

https://github.com/Automattic/studio/pull/1526/commits/bb9fa2d35669e829b52788411048d6cf2ef5749b#diff-4c2ef4a6b6d736371cd674652d469a629091efd56f2a31f04c1a72d9a477c158R80
